core/capabilities.go

```diff
@@ -0,0 +1,166 @@
+package core
+
+import (
+	"errors"
+
+	"github.com/livepeer/go-livepeer/drivers"
+	"github.com/livepeer/go-livepeer/net"
+	"github.com/livepeer/lpms/ffmpeg"
+)
+
+type Capability int
+type CapabilityString []uint64
+type Constraints struct{}
+type Capabilities struct {
+	bitstring   CapabilityString
+	constraints Constraints
+}
+
+const (
+	Capability_Invalid Capability = iota - 2
+	Capability_Unused
+	Capability_H264
+	Capability_MPEGTS
+	Capability_MP4
+	Capability_FractionalFramerates
+	Capability_StorageDirect
+	Capability_StorageS3
+	Capability_StorageGCS
```

Comment:
Figured it'd be good to be as specific as possible about all the features that are being used, including the storage driver, in case we ever need to disable support for one of these someday.